File: Frontend/material-dashboard-angular2/src/app/programming/interfaces/IUsuarioService.ts
import { Observable } from "rxjs";
import { Response } from "../models/Response.dto";

export interface IUsuarioService<T>{

    getAll(): Observable<Response<T[]>>;

    createOrUpdate(entity: T): Observable<Response<T>>;

    accesoExitoso(usuario: string, correo: string, clave: string): Observable<Response<T>>;

    recuperarContraseña(usuario: string, correo: string): Observable<Response<T>>;


}

File: Backend/app_administracion_medicamentos/src/main/java/app_administracion_medicamentos_backend/api/interfaces/services/IUsuarioService.java
package app_administracion_medicamentos_backend.api.interfaces.services;

import java.util.List;

public interface IUsuarioService<T> {

	
	public List<T> getAll();

	public T createOrUpdate(T entity);

	public T accesoExitoso(String usuario, String correo, String clave);
	
	public T recuperarClave(String usuario, String correo);
	
}
